core/controllers/acl_decorators.py

```diff
@@ -1983,8 +1983,9 @@ def test_can_delete_story(self, topic_id, **kwargs):
         if not self.user_id:
             raise base.UserFacingExceptions.NotLoggedInException
 
+        topic = topic_services.get_topic_by_id(topic_id, strict=False)
```

Comment:
Reasoning for this new check?

Reply:
This was due to the removal of the same check from one of the controllers, I think.
 Just as a safeguard if topic and topic rights ever get disconnected.